src/caip-types.test.ts

```diff
@@ -274,3 +278,46 @@ describe('parseCaipAccountId', () => {
     );
   });
 });
+
+describe('toCaipChainId', () => {
+  // This function relies on @metamask/utils CAIP helpers. Those are being
+  // tested with a variety of inputs.
+  // Here we mainly focus on our own wrapper around those:
+
+  it('returns a valid CAIP-2 chain ID', () => {
+    const namespace = KnownCaipNamespace.Eip155;
+    const reference = '1';
+    expect(toCaipChainId(namespace, reference)).toBe(
+      `${namespace}:${reference}`,
+    );
+  });
```

Comment:
Does `toCaipChainId` care whether the namespace of the chain ID matches a known namespace? It seems like it doesn't matter to me. So, do we need to use the values of `KnownCaipNamespace` to test `toCaipChainId` or is it sufficient to use a generic namespace? I'm thinking we might want something like this instead:

``` suggestion
  it('returns a valid CAIP-2 chain ID when given a valid namespace and reference', () => {
    const namespace = 'abc';
    const reference = '1';
    expect(toCaipChainId(namespace, reference)).toBe(
      `${namespace}:${reference}`,
    );
  });
```

That said, I do agree that anything in `KnownCaipNamespace` should be valid input to `toCaipChainId`. So we could add tests for it like this:

``` suggestion
  it('returns a valid CAIP-2 chain ID when given a valid namespace and reference', () => {
    const namespace = 'abc';
    const reference = '1';
    expect(toCaipChainId(namespace, reference)).toBe(
      `${namespace}:${reference}`,
    );
  });

  it.each(Object.values(KnownCaipNamespace))(
    'treats %s as a valid namespace',
    (namespace) => {
      const reference = '1';
      expect(toCaipChainId(namespace, reference)).toBe(
        `${namespace}:${reference}`,
      );
    }
  );
```

What do you think?

Reply:
I really love this idea yes! 😄 I'll keep this suggestion, thanks!